### ISSUE TITLE
listObjects(bucketName) returns 500 error (Possible AWS auth v4 issue)

### ISSUE THREAD

**Issue by Flajt** [xtyxtyx/minio-dart#31]
Hi, first of all thank you for the package.
I'm currently trying to run a `listObjects(myBucket)`, the issue is that I'm always getting an error back. I've tried it via aws cli, but there everything works without issue.
Maybe for debugging this could be helpfull:
**S3 endpoint**: s3.filebase.com 

I've also looked at both requests and found one maybe two possible differences between the library headers and the aws cli headers.

**Library Headers:**
```
REQUEST: GET https://s3.filebase.com/gs9omvzyaeobdt9fxipudyo6dna2?&prefix=&delimiter=%2F&list-type=2&maxKeys=1000
I/flutter (26600): host: s3.filebase.com
I/flutter (26600): user-agent: MinIO (Unknown; Unknown) minio-dart/2.0.0
I/flutter (26600): x-amz-date: 20210824T223349Z
I/flutter (26600): x-amz-content-sha256: UNSIGNED-PAYLOAD
I/flutter (26600): authorization: AWS4-HMAC-SHA256 Credential=>>MYKEY<</20210824/us-east-1/s3/aws4_request, 
SignedHeaders=host;x-amz-content-sha256;x-amz-date, Signature=>>A SIGNATURE<<
```
(the >>VARNAME<<, replaces the value, as I'm not sure if it's sensitive or not).

**AWS cli headers:**
``` 
REQUEST: GET https://s3.filebase.com/gs9omvzyaeobdt9fxipudyo6dna2?&prefix=&delimiter=%2F&list-type=2&maxKeys=1000
headers={'User-Agent': b'aws-cli/2.2.31 Python/3.8.8 Linux/5.4.0-81-generic exe/x86_64.ubuntu.20 prompt/off command/s3.ls', 
'X-Amz-Date': b'20210824T222928Z', 
'X-Amz-Content-SHA256': b'e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855', 
'Authorization': b'AWS4-HMAC-SHA256 Credential=>>MYKEY<</20210824/us-east-1/s3/aws4_request, SignedHeaders=host;x-amz-content-sha256;x-amz-date
, Signature=>>A DIFFERENT SIGNATURE<<
}
```
The first difference is the user agent, idk if that is important, some backends maybe have issues with that, but the second one is the **x-amz-content-sha256**: Which is UNSIGNED-PAYLOAD vs e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855 

Due to [this](https://stackoverflow.com/questions/39555841/when-calling-the-listobjects-operation-missing-required-header-for-this-request#39559909) stackoverflow posts  answer (the last one with 0 upvotes), I have the feeling that one might need to add this empty string hash/signature with the request instead of UNSIGNED-PAYLOAD. The empty string hash/signature of the answer matches with the one the cli provided. So I'm currently assuming that is the source of the error, but idk for sure.
 
 
Here also the error response, if needed, but I guess it's not really informative.
```
RESPONSE: 500 Internal Server Error
I/flutter (26600): connection: keep-alive
I/flutter (26600): date: Tue, 24 Aug 2021 22:34:04 GMT
I/flutter (26600): transfer-encoding: chunked
I/flutter (26600): x-runtime: 0.010055
I/flutter (26600): content-type: application/xml
I/flutter (26600): x-amz-request-id: ebd52b58-e1e3-4d46-9c0e-3a5f0f014617
I/flutter (26600): x-amz-id-2: ZmlsZWJhc2UtNThjODY1NGM2OS1ubm0yOQ==
I/flutter (26600): server: Filebase
I/flutter (26600): <?xml version="1.0" encoding="UTF-8"?><Error><Code>InternalError</Code><Message>We encountered an internal error. Please try again.</Message><RequestId>ebd52b58-e1e3-4d46-9c0e-3a5f0f014617</RequestId><HostId>ZmlsZWJhc2UtNThjODY1NGM2OS1ubm0yOQ==</HostId></Error> 
```
The devide I'm currently using is an Galaxy A51 with Android 11

After reading around I realized it might be an issue with AWS signature version 4, due to the fact that the storage provider only support AWS v4 signature requests.

**Comment by jamongeon1** [comment]
Hey, is there any update on this Issue?

**Comment by Flajt** [comment]
@jamongeon1 not as far as I know, I'm not sure if it's something specific to this library as I encoutered something like this in JS as well (maybe it's an issue with the S3 provider). But my knowledge with S3 is to limited to debug it myself. Might just me beeing stupid.

**Comment by xtyxtyx** [comment]
Thanks for your feedback!
This has been fixed in `3.1.0`.

**Comment by jamongeon1** [comment]
Thanks so much! The Filebase team appreciates you guys pushing a fix for this and adding us to the ReadME  😄
